Lib/test/test_ast/test_ast.py

```diff
@@ -814,6 +814,17 @@ def test_repr_large_input_crash(self):
                                     r"Exceeds the limit \(\d+ digits\)"):
             repr(ast.Constant(value=eval(source)))
 
+    def test_unparse_interactive(self):
+        # gh-129598: Fix of ast.unparse() when ast.Interactive contains multiple statements
+        source = "i = 1; 'expr'; raise Exception"
+        self.assertEqual(source, ast.unparse(ast.parse(source, mode='single')))
```

Comment:
Maybe also some test cases when we end the text with a semicolon, as well as when we end the text with a semicolon and a whitespace (I think we don't have a full roundtrip in this case).

Reply:
Won't do anything since semicolon info is not stored, just statements, so essentially parse just strips out the semi at the end.
```py
>>> unparse(parse('i = 1;'))
'i = 1'
>>> unparse(parse('i = 1;', mode='single'))
'i = 1'
>>> unparse(parse('i = 1; ', mode='single'))
'i = 1'
```

Reply:
Ok. Then no need for those test cases (unless we may want to add them just for robustness).